components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/SettingsMappingUtil.java

```diff
@@ -67,4 +71,27 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable) throws APIManageme
         }
         return scopeList;
     }
+
+    public List<String> GetRoleScopeList(String[] userRoles, String username) {
+        List<String> userRoleList;
+        List<String> authorizedScopes = new ArrayList<>();
+
+        if (userRoles == null || userRoles.length == 0) {
+            userRoles = new String[0];
+        }
+
+        userRoleList = Arrays.asList(userRoles);
+        Map<String, String> scopeRoleMapping =
+                APIUtil.getRESTAPIScopesForTenant(MultitenantUtils.getTenantDomain(username));
+        Iterator<Map.Entry<String, String>> iterator = scopeRoleMapping.entrySet().iterator();
```

Comment:
possible NPE

Reply:
Since we are getting all the scopes from the tenant and extract the scopes that mapped with the user's roles. There would be scope for all roles as per scopeIssuer. In this case, we won't get NPE.